crates/core/src/operations/connect.rs

```diff
@@ -712,9 +711,17 @@ pub(crate) async fn initial_join_procedure(
                 }
                 
                 let mut shuffled_gateways: Vec<_> = available_gateways.into_iter().collect();
-                use rand::seq::SliceRandom;
-                let mut rng = rand::thread_rng();
-                shuffled_gateways.shuffle(&mut rng);
+                let seed = std::time::SystemTime::now()
+                    .duration_since(std::time::UNIX_EPOCH)
+                    .unwrap_or_default()
+                    .as_secs();
+                
+                if !shuffled_gateways.is_empty() {
+                    for i in (1..shuffled_gateways.len()).rev() {
```

Comment:
mmm we don't need this logic in here polluting it, that's why we have the `shuffle` method